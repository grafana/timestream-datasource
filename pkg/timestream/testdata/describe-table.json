{
    "ColumnInfo": [
        {
            "Name": "Column",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "VARCHAR",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "Type",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "VARCHAR",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "Timestream attribute type",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "VARCHAR",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        }
    ],
    "NextToken": null,
    "QueryId": "#QueryId#",
    "QueryStatus": {
        "CumulativeBytesMetered": 0,
        "CumulativeBytesScanned": 0,
        "ProgressPercentage": 100
    },
    "Rows": [
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "availability_zone",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "microservice_name",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "hostname",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "instance_name",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "process_name",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "os_version",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "az",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "jdk_version",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "region",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "cell",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "silo",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "instance_type",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "DIMENSION",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "measure_name",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "MEASURE_NAME",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "time",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "timestamp",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "TIMESTAMP",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "measure_value::double",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "double",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "MEASURE_VALUE",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "measure_value::bigint",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "bigint",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "MEASURE_VALUE",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "measure_value::varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "varchar",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "MEASURE_VALUE",
                    "TimeSeriesValue": null
                }
            ]
        }
    ]
}