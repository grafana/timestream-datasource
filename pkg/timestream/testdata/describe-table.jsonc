//  ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ
//  
//  Frame[0] {
//      "typeVersion": [
//          0,
//          0
//      ]
//  }
//  Name: 
//  Dimensions: 3 Fields by 17 Rows
//  +-------------------+-----------------+---------------------------------+
//  | Name: Column      | Name: Type      | Name: Timestream attribute type |
//  | Labels:           | Labels:         | Labels:                         |
//  | Type: []*string   | Type: []*string | Type: []*string                 |
//  +-------------------+-----------------+---------------------------------+
//  | availability_zone | varchar         | DIMENSION                       |
//  | microservice_name | varchar         | DIMENSION                       |
//  | hostname          | varchar         | DIMENSION                       |
//  | instance_name     | varchar         | DIMENSION                       |
//  | process_name      | varchar         | DIMENSION                       |
//  | os_version        | varchar         | DIMENSION                       |
//  | az                | varchar         | DIMENSION                       |
//  | jdk_version       | varchar         | DIMENSION                       |
//  | region            | varchar         | DIMENSION                       |
//  | ...               | ...             | ...                             |
//  +-------------------+-----------------+---------------------------------+
//  
//  
//  ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ
{
  "status": 200,
  "frames": [
    {
      "schema": {
        "meta": {
          "typeVersion": [
            0,
            0
          ]
        },
        "fields": [
          {
            "name": "Column",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          },
          {
            "name": "Type",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          },
          {
            "name": "Timestream attribute type",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          }
        ]
      },
      "data": {
        "values": [
          [
            "availability_zone",
            "microservice_name",
            "hostname",
            "instance_name",
            "process_name",
            "os_version",
            "az",
            "jdk_version",
            "region",
            "cell",
            "silo",
            "instance_type",
            "measure_name",
            "time",
            "measure_value::double",
            "measure_value::bigint",
            "measure_value::varchar"
          ],
          [
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "varchar",
            "timestamp",
            "double",
            "bigint",
            "varchar"
          ],
          [
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "DIMENSION",
            "MEASURE_NAME",
            "TIMESTAMP",
            "MEASURE_VALUE",
            "MEASURE_VALUE",
            "MEASURE_VALUE"
          ]
        ]
      }
    }
  ]
}