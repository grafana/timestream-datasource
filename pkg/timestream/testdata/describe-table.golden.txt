ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ

Frame[0] {}
Name: 
Dimensions: 3 Fields by 17 Rows
+-------------------+-----------------+---------------------------------+
| Name: Column      | Name: Type      | Name: Timestream attribute type |
| Labels:           | Labels:         | Labels:                         |
| Type: []*string   | Type: []*string | Type: []*string                 |
+-------------------+-----------------+---------------------------------+
| availability_zone | varchar         | DIMENSION                       |
| microservice_name | varchar         | DIMENSION                       |
| hostname          | varchar         | DIMENSION                       |
| instance_name     | varchar         | DIMENSION                       |
| process_name      | varchar         | DIMENSION                       |
| os_version        | varchar         | DIMENSION                       |
| az                | varchar         | DIMENSION                       |
| jdk_version       | varchar         | DIMENSION                       |
| region            | varchar         | DIMENSION                       |
| ...               | ...             | ...                             |
+-------------------+-----------------+---------------------------------+


====== TEST DATA RESPONSE (arrow base64) ======
FRAME=QVJST1cxAAD/////AAIAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEDAAoADAAAAAgABAAKAAAACAAAAHQAAAADAAAATAAAACgAAAAEAAAAiP7//wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAACo/v//CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAMj+//8IAAAADAAAAAIAAAB7fQAABAAAAG1ldGEAAAAAAwAAAPQAAACIAAAABAAAACr///8UAAAAUAAAAFAAAAAAAAUBTAAAAAEAAAAEAAAAGP///wgAAAAkAAAAGQAAAFRpbWVzdHJlYW0gYXR0cmlidXRlIHR5cGUAAAAEAAAAbmFtZQAAAAAAAAAAJP///xkAAABUaW1lc3RyZWFtIGF0dHJpYnV0ZSB0eXBlAAAAqv///xQAAAA8AAAAPAAAAAAABQE4AAAAAQAAAAQAAACY////CAAAABAAAAAEAAAAVHlwZQAAAAAEAAAAbmFtZQAAAAAAAAAAkP///wQAAABUeXBlAAASABgAFAATABIADAAAAAgABAASAAAAFAAAAEQAAABIAAAAAAAFAUQAAAABAAAADAAAAAgADAAIAAQACAAAAAgAAAAQAAAABgAAAENvbHVtbgAABAAAAG5hbWUAAAAAAAAAAAQABAAEAAAABgAAAENvbHVtbgAA/////xgBAAAUAAAAAAAAAAwAFgAUABMADAAEAAwAAADAAgAAAAAAABQAAAAAAAADAwAKABgADAAIAAQACgAAABQAAACoAAAAEQAAAAAAAAAAAAAACQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEgAAAAAAAAASAAAAAAAAADIAAAAAAAAABABAAAAAAAAAAAAAAAAAAAQAQAAAAAAAEgAAAAAAAAAWAEAAAAAAAB4AAAAAAAAANABAAAAAAAAAAAAAAAAAADQAQAAAAAAAEgAAAAAAAAAGAIAAAAAAACoAAAAAAAAAAAAAAADAAAAEQAAAAAAAAAAAAAAAAAAABEAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAAAABEAAAAiAAAAKgAAADcAAABDAAAATQAAAE8AAABaAAAAYAAAAGQAAABoAAAAdQAAAIoAAACfAAAAtQAAAMEAAADFAAAAYXZhaWxhYmlsaXR5X3pvbmVtaWNyb3NlcnZpY2VfbmFtZWhvc3RuYW1laW5zdGFuY2VfbmFtZXByb2Nlc3NfbmFtZW9zX3ZlcnNpb25hempka192ZXJzaW9ucmVnaW9uY2VsbHNpbG9pbnN0YW5jZV90eXBlbWVhc3VyZV92YWx1ZTo6ZG91YmxlbWVhc3VyZV92YWx1ZTo6YmlnaW50bWVhc3VyZV92YWx1ZTo6dmFyY2hhcm1lYXN1cmVfbmFtZXRpbWUAAAAAAAAABwAAAA4AAAAVAAAAHAAAACMAAAAqAAAAMQAAADgAAAA/AAAARgAAAE0AAABUAAAAWgAAAGAAAABnAAAAbgAAAHcAAAB2YXJjaGFydmFyY2hhcnZhcmNoYXJ2YXJjaGFydmFyY2hhcnZhcmNoYXJ2YXJjaGFydmFyY2hhcnZhcmNoYXJ2YXJjaGFydmFyY2hhcnZhcmNoYXJkb3VibGViaWdpbnR2YXJjaGFydmFyY2hhcnRpbWVzdGFtcAAAAAAACQAAABIAAAAbAAAAJAAAAC0AAAA2AAAAPwAAAEgAAABRAAAAWgAAAGMAAABsAAAAeQAAAIYAAACTAAAAnwAAAKgAAABESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05ESU1FTlNJT05NRUFTVVJFX1ZBTFVFTUVBU1VSRV9WQUxVRU1FQVNVUkVfVkFMVUVNRUFTVVJFX05BTUVUSU1FU1RBTVAQAAAADAAUABIADAAIAAQADAAAABAAAAAsAAAAPAAAAAAAAwABAAAAEAIAAAAAAAAgAQAAAAAAAMACAAAAAAAAAAAAAAAAAAAAAAAAAAAKAAwAAAAIAAQACgAAAAgAAAB0AAAAAwAAAEwAAAAoAAAABAAAAIj+//8IAAAADAAAAAAAAAAAAAAABQAAAHJlZklkAAAAqP7//wgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAADI/v//CAAAAAwAAAACAAAAe30AAAQAAABtZXRhAAAAAAMAAAD0AAAAiAAAAAQAAAAq////FAAAAFAAAABQAAAAAAAFAUwAAAABAAAABAAAABj///8IAAAAJAAAABkAAABUaW1lc3RyZWFtIGF0dHJpYnV0ZSB0eXBlAAAABAAAAG5hbWUAAAAAAAAAACT///8ZAAAAVGltZXN0cmVhbSBhdHRyaWJ1dGUgdHlwZQAAAKr///8UAAAAPAAAADwAAAAAAAUBOAAAAAEAAAAEAAAAmP///wgAAAAQAAAABAAAAFR5cGUAAAAABAAAAG5hbWUAAAAAAAAAAJD///8EAAAAVHlwZQAAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABEAAAASAAAAAAABQFEAAAAAQAAAAwAAAAIAAwACAAEAAgAAAAIAAAAEAAAAAYAAABDb2x1bW4AAAQAAABuYW1lAAAAAAAAAAAEAAQABAAAAAYAAABDb2x1bW4AADACAABBUlJPVzE=
